testimony/__init__.py

```diff
@@ -450,6 +462,43 @@ def bugs_report(testcases):
     )
 
 
+def tags_report(testcases):
+    """Lists the test cases matching the input tag."""
+    result = {}
+    result['tagged_testcases'] = []
+    if not SETTINGS['input_tags']:
+        print 'Input tags required for this report.  See testimony --help'
+        sys.exit()
+    # Change the input tags to lower case
+    input_tags_list = [tag.lower() for tag in SETTINGS['input_tags']]
+    for _, tests in testcases.items():
+        for testcase in tests:
+            if testcase.tags:
+                # Split the test case tags with ',' and change to lowercase
+                # and trim empty spaces
+                testcase_tags_list = [
```

Comment:
If you change from [ to { then you will have a set comprehension. I think we are safe to do that since we are supporting python 2.7+